jsapp/js/components/formLanding.js

```diff
@@ -10,124 +10,173 @@ import sessionStore from 'js/stores/session';
 import PopoverMenu from 'js/popoverMenu';
 import LoadingSpinner from 'js/components/common/loadingSpinner';
 import InlineMessage from 'js/components/common/inlineMessage';
+import Checkbox from 'js/components/common/checkbox';
+import Icon from 'js/components/common/icon';
 import mixins from '../mixins';
 import {actions} from '../actions';
 import DocumentTitle from 'react-document-title';
 import CopyToClipboard from 'react-copy-to-clipboard';
-import {
-  MODAL_TYPES,
-  COLLECTION_METHODS,
-} from '../constants';
+import {MODAL_TYPES, COLLECTION_METHODS} from '../constants';
 import {ROUTES} from 'js/router/routerConstants';
-import {
-  formatTime,
-  notify,
-} from 'utils';
-import {
-  Link,
-} from 'react-router-dom';
+import {formatTime, notify} from 'utils';
+import {buildUserUrl, ANON_USERNAME} from 'js/users/utils';
+import {Link} from 'react-router-dom';
 import {withRouter} from 'js/router/legacy';
-import {userCan, userCanRemoveSharedProject} from 'js/components/permissions/utils';
+import {
+  userCan,
+  userCanRemoveSharedProject,
+} from 'js/components/permissions/utils';
+import permConfig from 'js/components/permissions/permConfig';
+import {PERMISSIONS_CODENAMES} from 'js/constants';
 
 const DVCOUNT_LIMIT_MINIMUM = 20;
+const ANON_CAN_ADD_PERM_URL = permConfig.getPermissionByCodename(
+  PERMISSIONS_CODENAMES.add_submissions
+).url;
 
 class FormLanding extends React.Component {
-  constructor(props){
+  constructor(props) {
     super(props);
     this.state = {
       selectedCollectMethod: COLLECTION_METHODS.offline_url.id,
       DVCOUNT_LIMIT: DVCOUNT_LIMIT_MINIMUM,
       nextPageUrl: null,
-      nextPagesVersions: []
+      nextPagesVersions: [],
+      anonymousSubmissions: false,
+      anonymousPermissions: [],
     };
     autoBind(this);
   }
-  componentDidMount () {
+  componentDidMount() {
     // reset loaded versions when new one is deployed
-    this.listenTo(actions.resources.deployAsset.completed, this.resetLoadedVersions);
+    this.listenTo(
+      actions.resources.deployAsset.completed,
+      this.resetLoadedVersions
+    );
+    this.listenTo(
+      actions.permissions.getAssetPermissions.completed,
+      this.onAssetPermissionsUpdated
+    );
+
+    actions.permissions.getAssetPermissions(this.props.params.uid);
+  }
+
+  onAssetPermissionsUpdated(response) {
+    const publicPerms = response.filter(
+      (assignment) => assignment.user === buildUserUrl(ANON_USERNAME)
+    );
+    const anonCanAdd = publicPerms.filter(
+      (perm) => perm.permission === ANON_CAN_ADD_PERM_URL
+    )[0];
+
+    this.setState({
+      anonymousPermissions: publicPerms,
+      anonymousSubmissions: anonCanAdd ? true : false,
+    });
+  }
+  updateAssetAnonymousSubmissions() {
+    const permission = this.state.anonymousPermissions.filter(
+      (perm) =>
+        perm.permission ===
+        permConfig.getPermissionByCodename(PERMISSIONS_CODENAMES.add_submissions).url
+    )[0];
```

Comment:
This was a direct copy from `publicSharePermissions.component.tsx`, but since I'm adding more logic here with a state (instead of getting it from props) I think changing it to `.find()` is a better solution 👍